utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/rules/DualScreenTestRule.kt

```diff
@@ -0,0 +1,143 @@
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                before(description)
+                try {
+                    base.evaluate()
+                } finally {
+                    after()
+                }
+            }
+        }
+    }
+
+    private fun before(description: Description) {
+        val dualScreenTestAnnotation = getAnnotation(description, DualScreenTest::class.java)
+        val singleScreenTestAnnotation = getAnnotation(description, SingleScreenTest::class.java)
+        val deviceOrientationAnnotation = getAnnotation(description, DeviceOrientation::class.java)
+        val requestedDeviceOrientation = requestedDeviceOrientation(
+            dualScreenTestAnnotation,
+            singleScreenTestAnnotation,
+            deviceOrientationAnnotation
+        )
+
+        val runDualScreenTest = dualScreenTestAnnotation != null
+        if (uiDevice.isSurfaceDuo()) {
+            if (runDualScreenTest) {
+                uiDevice.spanFromStart()
+            }
+        } else {
+            WindowInfoTracker.overrideDecorator(overrideServices)
+            mockFoldingFeature(requestedDeviceOrientation, runDualScreenTest)
+        }
+
+        rotateDevice(requestedDeviceOrientation)
+    }
+
+    private fun requestedDeviceOrientation(
+        dualScreenTestAnnotation: DualScreenTest?,
+        singleScreenTestAnnotation: SingleScreenTest?,
+        deviceOrientationAnnotation: DeviceOrientation?
+    ): Int {
+        var requestedDeviceOrientation = dualScreenTestAnnotation?.orientation ?: singleScreenTestAnnotation?.orientation
+        if (requestedDeviceOrientation == null || requestedDeviceOrientation == Int.MIN_VALUE) {
+            requestedDeviceOrientation = deviceOrientationAnnotation?.orientation ?: ROTATION_FREEZE_0
+        }
+        return requestedDeviceOrientation
+    }
+
+    private fun <T : Annotation> getAnnotation(description: Description, annotationClass: Class<T>): T? {
+        return description.testClass.getAnnotation(annotationClass)
+            ?: description.getAnnotation(annotationClass)
+    }
+
+    private fun rotateDevice(requestedDeviceOrientation: Int) {
+        when (requestedDeviceOrientation) {
+            ROTATION_FREEZE_270 -> uiDevice.setOrientationRight()
+            ROTATION_FREEZE_90 -> uiDevice.setOrientationLeft()
+            ROTATION_FREEZE_0 -> uiDevice.setOrientationNatural()
+        }
+    }
+
+    private fun mockFoldingFeature(deviceOrientation: Int, forDualScreenTest: Boolean) {
+        val displayFeatures = if (forDualScreenTest) {
+            val foldingFeature = FoldingFeature(
+                windowBounds = Rect(0, 0, uiDevice.displayWidth, uiDevice.displayHeight),
+                state = FoldingFeature.State.FLAT,
+                size = 0,
+                orientation = getFoldingFeatureOrientation(deviceOrientation)
```

Comment:
Maybe for a future version, we can add folding feature properties as parameters to the Test Annotation - there may be some cases where developers want specific properties

Also, was there a reason that `FoldingFeature.State.FLAT` was chosen as the default value? If not, I would suggest updating it to `HALF_OPEN`, just in case some app logic depends on `isSeparating` being true for dual-screen UI to appear

Reply:
Changed to FoldingFeature.State.HALF_OPENED